src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -169,15 +167,32 @@ namespace CommandLineInterface {
         opt.getUsage(usage);
 
         // Set path of EnergyPlus program path
-        exeDirectory = getParentDirectoryPath(getAbsolutePath(getProgramPath()));
-
-        opt.get("-w")->getString(state.files.inputWeatherFileName.fileName);
+        DataStringGlobals::exeDirectoryPath = FileSystem::getParentDirectoryPath(FileSystem::getAbsolutePath(FileSystem::getProgramPath()));
+
+        // ezOptionParser doesn't have a getPath, so we get a string, then convert it to path
+        // I'm limiting the scope of the std::string temporaries I used to avoid mistakes
+        {
+            std::string inputWeatherFileName;
+            opt.get("-w")->getString(inputWeatherFileName);
+            state.files.inputWeatherFilePath.filePath = fs::path{inputWeatherFileName};
+        }
 
-        opt.get("-i")->getString(inputIddFileName);
+        {
+            // TODO: should this vbe in IOFiles as an InputFile?
+            std::string inputIddFileName;
+            opt.get("-i")->getString(inputIddFileName);
+            DataStringGlobals::inputIddFilePath = fs::path{inputIddFileName};
+        }
 
-        if (!opt.isSet("-i") && !legacyMode) inputIddFileName = exeDirectory + inputIddFileName;
+        if (!opt.isSet("-i") && !legacyMode) {
+            DataStringGlobals::inputIddFilePath = DataStringGlobals::exeDirectoryPath / DataStringGlobals::inputIddFilePath;
```

Comment:
:heart_eyes: I know it's silly, but this is just nice.  It's the same as when I started using `pathlib` in Python.  This just makes so much sense.